providers/snowflake/snowflake_provider.go

```diff
@@ -0,0 +1,97 @@
+// Copyright 2019 The Terraformer Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package snowflake
+
+import (
+	"errors"
+
+	"github.com/GoogleCloudPlatform/terraformer/terraform_utils"
+	"github.com/kelseyhightower/envconfig"
+)
+
+type SnowflakeConfig struct {
+	Account  string `envconfig:"SNOWFLAKE_ACCOUNT" required:"true"`
+	Username string `envconfig:"SNOWFLAKE_USER" required:"true"`
+	Password string `envconfig:"SNOWFLAKE_PASSWORD" required:"true"`
+	Region   string `envconfig:"SNOWFLAKE_REGION" default:""`
+	Role     string `envconfig:"SNOWFLAKE_ROLE" required:"true"`
+}
+
+type SnowflakeProvider struct {
+	terraform_utils.Provider
+
+	Config *SnowflakeConfig
+}
+
+func (p *SnowflakeProvider) Init(args []string) error {
+	p.Config = &SnowflakeConfig{}
+	err := envconfig.Process("", p.Config)
+	if err != nil {
+		return err
+	}
+
+	// us-west-2 is their default region, but if you actually specify that it won't trigger their default code
+	//  https://github.com/snowflakedb/gosnowflake/blob/52137ce8c32eaf93b0bd22fc5c7297beff339812/dsn.go#L61
```

Comment:
do you want to open a bug on snowflakedb/gosnowflake to see if they add this code to their package?

Reply:
https://github.com/snowflakedb/gosnowflake/issues/270